cbor/shared/src/main/scala/fs2/data/cbor/low/internal/ItemParser.scala

```diff
@@ -42,7 +42,7 @@ private[low] object ItemParser {
       onEos: => Pull[F, CborItem, T]): Pull[F, CborItem, T] =
     if (idx >= chunk.size) {
       Pull.output(Chunk.seq(chunkAcc.reverse)) >> rest.pull.uncons.flatMap {
-        case Some((hd, tl)) => cont(hd, 0, tl, Nil)
+        case Some((hd, tl)) => ensureChunk(hd, 0, tl, Nil)(cont)(onEos)
```

Comment:
One thing I started wondering: Is it safe to assume `idx=0` here? Or could it be that we need to do something like `idx - chunk.size` here to be safe (for future developments)?

Reply:
This is an interesting topic, that would allow to have built-in selector in the parser. However, this will required to rethink the entire parser, so I prefer to not do it now, and think about it as a new implementation in the future.